tartiflette_plugin_scalars/duration.py

```diff
@@ -0,0 +1,72 @@
+from datetime import timedelta
+from typing import Union
+
+from tartiflette.constants import UNDEFINED_VALUE
+from tartiflette.language.ast import StringValueNode
+
+
+def _parse_duration(value: str) -> timedelta:
+    def get_kv(_arg: str) -> dict:
+        try:
+            _key, _value = _arg.split("=")
+        except ValueError:
+            raise ValueError(
+                f"Duration argument has more or less than 2 elements: < {_arg} >"
+            )
+        else:
+            if _key in VALID_KEYS:
+                try:
+                    return {_key: int(_value)}
+                except ValueError:
+                    raise ValueError(
+                        f"Duration argument value is not an int: < {_arg} >"
+                    )
+            else:
+                raise ValueError(f"Duration argument has invalid key: < {_arg} >")
+
+    VALID_KEYS = (
+        "days",
+        "seconds",
+        "microseconds",
+        "milliseconds",
+        "minutes",
+        "hours",
+        "weeks",
+    )
+    # make a timedelta from a comma separated string
+    # determine if multiple args
+    value = value.replace(" ", "")
+    if "," in value:
+        # multiple
+        arg_list = value.split(",")
+        arg_dict = {}
+        for arg in arg_list:
+            if "=" in arg:
+                arg_dict.update(get_kv(arg))
+            else:
+                raise ValueError(f"Duration key missing '=': < {value} >")
+        return timedelta(**arg_dict)
+    else:
+        # single
+        if "=" in value:
```

Comment:
I'm pretty sure there's ways you could simplify this, because `value.split(",")` can work even on a single value.

Reply:
Committed an attempt to simplify it :)